ldi-core/ldes-client-v2/tree-node-fetcher/src/main/java/ldes/client/treenodefetcher/domain/valueobjects/MutabilityStatus.java

```diff
@@ -0,0 +1,38 @@
+package ldes.client.treenodefetcher.domain.valueobjects;
+
+import java.time.LocalDateTime;
+import java.util.Arrays;
+import java.util.Optional;
+
+public class MutabilityStatus {
+	private final boolean mutable;
+	private final LocalDateTime earliestNextVisit;
+
+	public MutabilityStatus(boolean mutable, LocalDateTime earliestNextVisit) {
+		this.mutable = mutable;
+		this.earliestNextVisit = earliestNextVisit;
+	}
+
+	public static MutabilityStatus ofHeader(Optional<String> cacheControlHeader) {
```

Comment:
passing an optional as a parameter is usually not considered good practice as you force everyone that calls this method, to wrap their input in an Optional. Unwrapping an optional en determining the flow should be the responsibility of the caller. A clean way (imo) would be two have two static ctr methods, one that expects a notNull cacheControlHeader and one without param. Currently you call this method like this:

``MutabilityStatus mutabilityStatus = MutabilityStatus.ofHeader(response.getValueOfHeader(HttpHeaders.CACHE_CONTROL));``

You could write it like: 
``MutabilityStatus mutabilityStatus = response.getValueOfHeader(HttpHeaders.CACHE_CONTROL).map(MutabilityStatus::ofHeader).orElse(MutabilityStatus::empty);``